hayeon/programmers-1/명예의-전당.js

```diff
@@ -0,0 +1,25 @@
+function solution(k, score) {
+  let stack = [];
+  let result = [];
+
+  for (let i = 0; i < score.length; i++) {
+    stack.push(score[i]);
+
+    if (stack.length >= k) {
+      stack.sort((a, b) => b - a);
+      result.push(stack[k - 1]);
+      continue;
+    }
+
+    result.push(Math.min(...stack));
+  }
+
+  return result;
+}
```

Comment:
```suggestion
function solution(k, score) {
  const stack = [];
  const result = [];

  for (let i = 0; i < score.length; i++) {
    stack.push(score[i]);

    if (stack.length >= k) {
      stack.sort((a, b) => b - a);
      result.push(stack[k - 1]);
      continue;
    }

    result.push(Math.min(...stack));
  }

  return result;
}
```
- 풀이하느라 고생하셧씁니다 ~ 🙇‍♂️
- stack, result 모두 const로 선언해주세요!
- `콜라 문제도 const를 말씀드렸는데 리마인드 드릴게요!!` ~ 스터디 초기때부터 강조드렸지만 변수를 특정 값으로 `재할당`이 아니면 `const`로 선언해서 사이드 이펙트를 방지해주세요!
- 아래처럼 reduce를 이용한 선언적으로 작성함으로써 좀 더 깔끔한 코드가 가능합니다 :)
```js
function solution(k, score) {
  const hallOfFame = [];

  return score.reduce((acc, cur) => {
    hallOfFame.push(cur);
    hallOfFame.sort((a, b) => b - a);

    hallOfFame.length > k && hallOfFame.pop();

    return [...acc, hallOfFame.at(-1)];
  }, []);
}
```

Reply:
오 reduce로도 가능하군요!!!!! reduce이용한 풀이로도 다시 한 번 봐야겠네요.. const는 다시 한번 새기겠습니다!!!!